setup.py

```diff
@@ -2,241 +2,62 @@
-        "PKGBUILD.template",
-        "scripts/*",
-        "docs/*",
-        "glade",
-        "test",
-        "ulauncher.desktop.dev",
-        "requirements.txt",
-    ])
-
-    DistUtilsExtra.auto.setup(
-        name='ulauncher',
-        version='%VERSION%',
-        license='GPL-3',
-        author='Aleksandr Gornostal',
-        author_email='ulauncher.app@gmail.com',
-        description='Application launcher for Linux',
-        url='https://ulauncher.io',
-        data_files=DataFileList([
-            ('share/icons/hicolor/48x48/apps', [
-                'data/media/icons/hicolor/ulauncher.svg'
-            ]),
-            ('share/icons/hicolor/48x48/apps', [
-                'data/media/icons/hicolor/ulauncher-indicator.svg'
-            ]),
-            ('share/icons/hicolor/scalable/apps', [
-                'data/media/icons/hicolor/ulauncher.svg'
-            ]),
-            ('share/icons/hicolor/scalable/apps', [
-                'data/media/icons/hicolor/ulauncher-indicator.svg'
-            ]),
-            # for fedora + GNOME
-            ('share/icons/gnome/scalable/apps', [
-                'data/media/icons/hicolor/ulauncher.svg'
-            ]),
-            ('share/icons/gnome/scalable/apps', [
-                'data/media/icons/hicolor/ulauncher-indicator.svg'
-            ]),
-            # for ubuntu
-            ('share/icons/breeze/apps/48', [
-                'data/media/icons/ubuntu-mono-light/ulauncher-indicator.svg'
-            ]),
-            ('share/icons/ubuntu-mono-dark/scalable/apps', [
-                'data/media/icons/ubuntu-mono-dark/ulauncher-indicator.svg'
-            ]),
-            ('share/icons/ubuntu-mono-light/scalable/apps', [
-                'data/media/icons/ubuntu-mono-light/ulauncher-indicator.svg'
-            ]),
-            ('share/icons/elementary/scalable/apps', [
-                'data/media/icons/elementary/ulauncher-indicator.svg'
-            ]),
-        ]),
-        cmdclass={'install': InstallAndUpdateDataDirectory}
-    )
-
-    # unpatch distutils-extra src_find
-    DistUtilsExtra.auto.src_find = original_find_src
-
-
-if __name__ == '__main__':
-    main()
+        # Build Preferences before python package build
+        subprocess.run(["./ul", "build-preferences"], check=True)
+        build_py.run(self)
+        print("Overwriting the namespace package with fixed values")
+        namespace_package = os.path.realpath(os.path.join(self.build_lib, "ulauncher/__init__.py"))
+        file = open(namespace_package, "w")
+        file.write("\n".join([
+            "__data_directory__ = '%s'" % os.path.join(sys.prefix, "share/ulauncher"),
+            "__version__ = '%s'" % __version__
+        ]))
+        file.close()
+
+
+setup(
+    packages=find_packages(exclude=["tests"]),
+    # These will be placed in /usr
+    data_files=[
+        ("share/applications", ["ulauncher.desktop"]),
+        ("lib/systemd/user", ["contrib/systemd/ulauncher.service"]),
```

Comment:
Looks like this file doesn't make it into the tarball either:
```
running install_data
error: can't copy 'contrib/systemd/ulauncher.service': doesn't exist or not a regular file
```

Reply:
That error says you [don't have](https://github.com/pypa/setuptools/blob/a4dbe3457d89cf67ee3aa571fdb149e6eb544e88/setuptools/_distutils/file_util.py#L103-L105) `contrib/systemd/ulauncher.service`, but it's in the source code. So I don't understand how you can't have it. I am running ulauncher from an installation with my setup.py rewrite now, and it includes the systemd unit, so at least this is working great for me.

Reply:
This is from the tarball (generated via `./ul`) not in the source directory.

Reply:
> This is from the tarball (generated via `./ul`) not in the source directory.

Thank you. I'll have to fix that (and should have tested that workflow).

Reply:
ulauncher.service is included in the tarball now, and setup.py no longer wraps ./ul, or needs the scripts directory. I would be surprised if actually works all the way through to build the Fedora package though.